api/v1alpha1/virtualmachinepublishrequest_types.go

```diff
@@ -179,7 +184,7 @@ type VirtualMachinePublishRequestSpec struct {
 	// will be marked in error.
 	//
 	// +optional
-	Target VirtualMachinePublishRequestTarget `json:"target,omitempty"`
+	Target *VirtualMachinePublishRequestTarget `json:"target,omitempty"`
```

Comment:
Out of curiosity, is the purpose of making these pointers so that you can check for `nil` values?  Even after marking these as `optional` and `omitempty`, do we still need these as pointers?

Reply:
I also prefer to avoid pointers wherever necessary as it avoids NPEs and is more efficient memory management. Yi is hitting a JSON patch error when these are not pointers, but frankly, we shouldn't be mutating the spec here anyway IMO. The intention was not to default via a mutation webhook.

Reply:
Switched back to use struct.